WebRoot/js/zimbraAdmin/mobileDevices/model/ZaRegisterDevice.js

```diff
@@ -0,0 +1,207 @@
+
+    return AjxMessageFormat.format(ZaMsg["TTL_Retention_Policy_d"], diffDays);
+}
+
+ZaRegisterDevice.getFilteredDevices = function (value) {
+    var result = ZaRegisterDevice.getRegisteredDevices("name", value);
+
+    return result;
+}
+
+ZaRegisterDevice.getRegisteredDevices =
+function(by, val) {
+    var soapDoc = AjxSoapDoc.create("GetDeviceStatusRequest",ZaZimbraAdmin.URN, null);
+
+    if (by && val) {
+       var account = soapDoc.set("account", val);
+        account.setAttribute("by", by);
+    }
+
+    var params = new Object();
+    params.soapDoc = soapDoc;
+
+    try{
+        var reqMgrParams = {
+            controller : ZaApp.getInstance().getCurrentController(),
+            busyMsg : ZaMsg.BUSY_GETTING_SYNC_DEVICES
+        };
+
+        var resp = ZaRequestMgr.invoke(params, reqMgrParams).Body.GetDeviceStatusResponse.device;
+        return resp;
+    } catch(ex) {
+        throw ex;
+        return null;
+    }
+}
+
+
+ZaRegisterDevice.quarantineDevice = function(obj) {
+    var soapDoc = AjxSoapDoc.create("QuarantineDeviceRequest",ZaZimbraAdmin.URN, null);
+
+    var device = soapDoc.set("device", null, null);
+    device.setAttribute("id", obj.id);
+    
+    var account = soapDoc.set("account", obj[ZaRegisterDevice.RD_Email_Address]);
+    account.setAttribute("by", "name");
+    // obj.status = 2;
+    // return [obj];
+    var params = new Object();
+    params.soapDoc = soapDoc;
+
+    try{
+        var reqMgrParams = {
+            controller : ZaApp.getInstance().getCurrentController(),
+            busyMsg : ZaMsg.BUSY_QUARANTINE_SYNC_DEVICES
+        };
+
+        var resp = ZaRequestMgr.invoke(params, reqMgrParams).Body.QuarantineDeviceResponse.device;
+        return resp;
+    } catch(ex) {
+        throw ex;
+        return null;
+    }
+}
+
+ZaRegisterDevice.removeDevice = function(obj) {
+    var soapDoc = AjxSoapDoc.create("RemoveDeviceRequest",ZaZimbraAdmin.URN, null);
+
+    var device = soapDoc.set("device", null, null);
+    device.setAttribute("id", obj.id);
+    
+    var account = soapDoc.set("account", obj[ZaRegisterDevice.RD_Email_Address]);
+    account.setAttribute("by", "name");
+    
+    var params = new Object();
+    params.soapDoc = soapDoc;
+
+    try{
+        var reqMgrParams = {
+            controller : ZaApp.getInstance().getCurrentController(),
+            busyMsg : ZaMsg.BUSY_REMOVING_SYNC_DEVICES
+        };
+
+        var resp = ZaRequestMgr.invoke(params, reqMgrParams).Body.RemoveDeviceResponse;
+
+    } catch(ex) {
+        throw ex;
+        return null;
+    }
+}
+
+ZaRegisterDevice.allowDeviceSync = function(obj) {
+    var soapDoc = AjxSoapDoc.create("AllowDeviceRequest",ZaZimbraAdmin.URN, null);
+
+    var device = soapDoc.set("device", null, null);
+    device.setAttribute("id", obj.id);
+    
+    var account = soapDoc.set("account", obj[ZaRegisterDevice.RD_Email_Address]);
+    account.setAttribute("by", "name");
+    
+    // obj.status = 0;
+    // return [obj];
```

Comment:
Remove commented code.

Reply:
removed was added for testing in local vm